corehq/apps/domain/views/settings.py

```diff
@@ -271,6 +273,103 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+@method_decorator([
+    domain_admin_required,
+    use_bootstrap5,
+    toggles.IP_ACCESS_CONTROLS.required_decorator(),
+], name='dispatch')
+class EditIPAccessConfigView(BaseProjectSettingsView):
+    template_name = 'domain/admin/ip_access_config.html'
+    urlname = 'ip_access_config'
+    page_title = gettext_lazy("IP Access")
+
+    @property
+    @memoized
+    def form(self):
+        initial = {}
+        domain_config = self.ip_access_config
+        if domain_config:
+            display_spacer = ", "
+            initial.update({
+                'country_allowlist': domain_config.country_allowlist,
+                'ip_allowlist': display_spacer.join(domain_config.ip_allowlist),
+                'ip_denylist': display_spacer.join(domain_config.ip_denylist),
+                'comment': domain_config.comment
+            })
+
+        if self.request.method == 'POST':
+            return IPAccessConfigForm(self.request.POST, initial=initial)
+        return IPAccessConfigForm(initial=initial)
+
+    @cached_property
+    def ip_access_config(self):
+        try:
+            return IPAccessConfig.objects.get(domain=self.domain)
+        except IPAccessConfig.DoesNotExist:
+            return None
+
+    @cached_property
+    def ip_country(self):
+        if settings.MAXMIND_LICENSE_KEY:
+            return get_ip_country(self.current_ip)
+        return None
+
+    @property
+    def current_ip(self):
+        return get_ip(self.request)
+
+    @property
+    def page_context(self):
+        return {
+            'ip_access_config_form': self.form,
+        }
+
+    def post(self, request, *args, **kwargs):
+        if self.form.is_valid():
```

Comment:
Can you move some/all of the below validation checks into the form, so validation is all in one place?  That'd also let errors be attached to the associated fields automatically.

Reply:
It wouldn't give as detailed a message, but you could also just do a generic "will this config let me access the project" check like so:
```python
IPAccessConfig(
    country_allowlist=form.cleaned_data['country_allowlist'],
    ip_allowlist=form.cleaned_data['ip_allowlist'],
    ip_denylist=form.cleaned_data['ip_denylist'],
).is_allowed(get_ip(request))
```
That would ensure the validation matches the implementation exactly